llvm/lib/CodeGen/MachineSink.cpp

```diff
@@ -799,6 +816,30 @@ bool MachineSinking::runOnMachineFunction(MachineFunction &MF) {
     }
   }
 
+  if (AggressivelySinkInstsIntoCycle) {
```

Comment:
Integrate this into the above loop in SinkInstsIntoCycle? I don't see why it's a second round of post-processing the main sinking

Reply:
The idea was to allow for a sinking strategy of increasing aggressiveness based on RP. But, I don't think we're close to having that sort of functionality, and this probably isn't the correct avenue for a feature of that sort to begin with.

Reply:
On second thought, I think the improved sinking capability would only make sense in the context of something like this.